File: Advent3/src/main/python/Advent2024/D01.py
inp = """3   4
4   3
2   5
1   3
3   9
3   3"""

lines = [[int(s) for s in ln.split()] for ln in inp.splitlines()]
l, r = zip(*lines)
l = sorted(l)
r = sorted(r)

print(sum(abs(x - y) for x,y in zip(l, r)))

File: Advent3/src/main/scala/Advent2024/D01.scala
package Advent2024

import Shared.D

private[this] object D01 extends D {
  val List(l, r) = Input.lines.map(s => raw"\s+".r.split(s).map(_.toInt)).transpose.map(_.sorted)
  l.zip(r).map { case (x, y) => Math.abs(x - y) }.sum.part
  val rFreq = r.freqMap
  l.map { x => x * rFreq.getOrElse(x, 0) }.sum.part
}
